dev/get_changes.py

```diff
@@ -0,0 +1,134 @@
+            
+    def describe_changes(self):
+        for pull in self.pulls:
+
+            """ Assembles change description with PR and user URLs """
+            entry = []
+            if pull.user.login != 'dependabot[bot]':
+                labels = []
+                for label in pull.labels:
+                    if label.name != 'documentation':
+                        labels.append(label.name)
+                change_str = f'* **{labels[0]}: {pull.title}** [`#{pull.number}`]({pull.html_url}) [@{pull.user.login}]({pull.user.html_url})  '
+                
+                """ Extracts one-line description if present """
+                beg = pull.body.find('One-line summary:') + 18
+                if beg == 17:
+                    change_descrip_str = '    **'
+                else:
+                    test = pull.body.find('### Checklist')
+                    if test == -1:
+                        end = beg + 75
+                    else:
+                        end = test - 1
+                    descrip = pull.body[beg:end].split()
+                    descrip_str = ' '.join(descrip)
+                    change_descrip_str = f'    *{descrip_str}*'
+                
+                entry.append(change_str + '\n')
+                entry.append(change_descrip_str + '\n')
+                self.text.append(entry)
+
+    def get_new_contributors(self):
+        for pull in self.pulls:
+            comments = pull.get_issue_comments()
+            for comment in comments:
+                if 'Thanks for opening your' in comment.body:
+                    self.new_contributors[pull.user.login] = pull.user.url
+        if self.new_contributors:
+            print('New contributors:')
+            for k, v in self.new_contributors.items():
+                print(f'@{k}: {v}')
+        else:
+            print('Note: no new contributors were identified.')
+
+    def update_changelog(self):
+        f = open('changes.txt', 'w+')
+        f = open('changes.txt', 'a')
+        f.write(self.rel_title_str + '\n')
+        for entry in self.text:
+            for line in entry:
+                f.write(line)
+        f.close()
+
+        with open('changes.txt', 'r+') as f:
+            new_changes = f.read()
+        with open(self.path, 'r') as contents:
+            save = contents.read()
+        with open(self.path, 'w') as contents:
+            contents.write(new_changes)
+        with open(self.path, 'a') as contents:
+            contents.write(save)
+
+@click.command()
+@click.option(
+    '--github_token', type=str, default=''
+)
+@click.option(
+    "--previous", type=str, default=''
+)
+@click.option(
+    "--current", type=str, default=''
+)
+@click.option(
+    "--path", 
+    type=str, 
+    default='/Users/michael/Documents/astronomer_projects/openlineage_upstream/OpenLineage/CHANGELOG.md',
```

Comment:
🙂

Reply:
Can we make this relative path?

Reply:
Unfortunately no -- relative paths aren't supported by the library. But I can remove the default.

Reply:
Despite the docs saying an absolute path is required, a relative path works. I tested it and made it the default.